toolchain/lowering/lowering_context.h

```diff
@@ -65,6 +64,15 @@ class LoweringContext {
   // caller.
   auto BuildType(SemanticsNodeId node_id) -> llvm::Type*;
 
+  // Returns the empty LLVM struct type used to represent the type `type`.
+  auto GetTypeType() -> llvm::StructType* {
+    if (!type_type_) {
+      // `type` is lowered to an empty LLVM StructType.
+      type_type_ = llvm::StructType::create(*llvm_context_, {}, "type");
```

Comment:
Naming this has the maybe-good and maybe-bad effect of causing each IR module to get a *different* type for `type`, and LLVM to not unique them... Really no idea whether this is good or bad. Anyways, probably not worth worrying about, and I can see why you prefer it from a readability perspective.

I think I actually argued for the prior behavior of using `{}` in LLVM IR for this, but the test case changes below seem pretty compelling so I think if so, I was wrong. =]